pkg/hope/hypervisors/esxi_hypervisor.go

```diff
@@ -18,11 +19,32 @@ import (
 
 type EsxiHypervisor struct {
 	node hope.Node
+
+	insecure bool
 }
 
 func (hyp *EsxiHypervisor) Initialize(node hope.Node) error {
 	hyp.node = node
-	return nil
+
+	errs := []error{}
+	for _, s := range node.Parameters {
```

Comment:
I wonder if this ends up simplifying if it starts with building a parameter map, and then picks away at the values until there are either none left, or a set of un-usable keys?